File: multi-frontend-backend-cloud/applications/backend/java-greeting/src/main/java/com/greeting/controller/GreetingController.java
package com.greeting.controller;

import com.greeting.model.GreetingModel;
import com.greeting.service.GreetingService;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;
import java.util.Optional;

@RestController
@RequestMapping("/greetings")
@AllArgsConstructor
public class GreetingController {

    private final GreetingService greetingService;

    @GetMapping()
    public List<GreetingModel> getGreetings() throws Exception {
        return greetingService.getGreetings();
    }

//    @GetMapping("/{language}")
//    public ResponseEntity<?> getFamily(@PathVariable("language") String language) {
//        final String languageCapitalized = language.substring(0, 1).toUpperCase() + language.substring(1);
//        final Optional<Map<String, String>> family = Optional.ofNullable(Data.families.get(languageCapitalized));
//        return family.isPresent() ? ResponseEntity.ok(family.get()) : ResponseEntity.badRequest().body("Not found");
//    }
}

File: multi-frontend-backend-cloud/applications/frontend/angularjs-s3/src/app/greeting/greeting.controller.ts
import { GreetingService } from './greeting.service';
import { Greetings } from './greeting.interfaces';

export class GreetingController {

  public greetings: Greetings;
  public loading: boolean = false;
  public spinnerDiameter: number = 40;

  constructor(private greetingService: GreetingService) {
    'ngInject';
  }

  public loadData(): void {
    this.loading = true;

    this.greetingService.getData().then((data: Greetings) => {
      this.greetings = data;
      console.log(data);
    }).finally(() => {
      this.loading = false;
    });
  }

}
